addons/udes_stock/tests/test_stock_move_line.py

```diff
@@ -0,0 +1,308 @@
+        self.assertEqual(quantity, 6, 'Expected the un-met quantity to be six')
+        self.assertIsNone(new_ml)
+
+    def test16_move_line_for_qty_split_success(self):
+        """ Test when a subset of move lines are to be returned, with one being split, and one original one
+            Uses the default sorting
+        """
+        # Create additional move lines
+        pick_info = [{'product': self.banana, 'qty': 17}]
+        self.create_picking(
+            self.picking_type_goods_in, products_info=pick_info, confirm=True, assign=True
+        )
+        result, new_ml, quantity = (
+            self.env['stock.move.line']
+            .search([('product_id.id', '=', self.banana.id)])
+            .move_lines_for_qty(19)
+        )
+        # Check returned moveline matches requested quantity
+        self.assertEqual(result.mapped('product_id'), self.banana, 'Expected product was a banana')
+        self.assertEqual(sum(result.mapped('product_qty')), 19, 'Expeceted quantity to be 19')
+        self.assertEqual(new_ml.mapped('product_id'), self.banana, 'Expected product was a banana')
+        self.assertEqual(
+            sum(new_ml.mapped('product_qty')), 3, 'Expeceted quantity of new ml was three'
+        )
+        self.assertEqual(quantity, 0, 'Expected the un-met quantity to be 0')
+        # Check that the largest move line is not split, and  smaller ones are
+        self.assertEqual(result.mapped('product_qty'), [17.0, 2.0])
+
+    def test17_move_line_for_qty_split_success(self):
+        """ Test when a subset of move lines are to be returned, with one being split, and one original one
+            Does not use the default sorting
+        """
+        # Create additional move lines
+        pick_info = [{'product': self.banana, 'qty': 17}]
+        self.create_picking(
+            self.picking_type_goods_in, products_info=pick_info, confirm=True, assign=True
+        )  # Sort the move lines in this case to ensure the smallest is first
+        mls = self.env['stock.move.line'].search([('product_id.id', '=', self.banana.id)])
+        sorted_mls = mls.sorted(lambda ml: ml.product_qty)
+        result, new_ml, quantity = sorted_mls.move_lines_for_qty(19, sort=False)
+        # Check returned moveline matches requested quantity
+        self.assertEqual(result.mapped('product_id'), self.banana, 'Expected product was a banana')
+        self.assertEqual(sum(result.mapped('product_qty')), 19, 'Expeceted quantity to be 19')
+        self.assertEqual(new_ml.mapped('product_id'), self.banana, 'Expected product was a banana')
+        self.assertEqual(
+            sum(new_ml.mapped('product_qty')), 3, 'Expeceted quantity of new ml was three'
+        )
+        self.assertEqual(quantity, 0, 'Expected the un-met quantity to be 0')
+        # Check that the small move line is not split
+        self.assertEqual(result.mapped('product_qty'), [5.0, 14.0])
+
+    def test18_move_line_for_qty_simple_success(self):
```

Comment:
This is the same as the previous `XXX_move_line_for_qty_simple_success` but with banana. Removed.